samila/__main__.py

```diff
@@ -1,11 +1,105 @@
 # -*- coding: utf-8 -*-
 """Samila main."""
 
+import argparse
+import matplotlib.pyplot as plt
+
 from art import tprint
-from .params import SAMILA_VERSION
+from .params import SAMILA_VERSION, GenerateMode, Projection, Marker
 from .functions import samila_help
+from .genimage import GenerativeImage
+
+
+def main():
+    """
+    CLI main function.
+
+    :return: None
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--version', help='version', action='store_true', default=False)
+
+    parser.add_argument('--load-config', help='load config', type=str)
+    parser.add_argument('--load-data', help='load data', type=str)
+    parser.add_argument('--function1', help='function1', type=str)
+    parser.add_argument('--function2', help='function2', type=str)
+    parser.add_argument('--function_seed', help='function seed', type=str)
+
+    parser.add_argument('--seed', help='seed', type=str)
+    parser.add_argument('--start', help='start', type=float)
+    parser.add_argument('--step', help='step', type=float)
+    parser.add_argument('--stop', help='stop', type=float)
+    parser.add_argument(
+        '--mode',
+        help='generation mode',
+        type=str,
+        choices=[x.value for x in GenerateMode])
+
+    parser.add_argument('--color', help='color', type=str)
+    parser.add_argument('--bgcolor', help='bgcolor', type=str)
+    parser.add_argument('--cmap', help='cmap', type=str)
+    parser.add_argument('--spot-size', help='spot size', type=float)
+    parser.add_argument('--size', help='size', type=tuple)
+    parser.add_argument('--alpha', help='alpha', type=float)
+    parser.add_argument('--linewidth', help='linewidth', type=float)
+    parser.add_argument('--rotation', help='rotation', type=float)
+    parser.add_argument(
+        '--projection',
+        help='projection type',
+        type=str,
+        choices=[x.value for x in Projection])
+    parser.add_argument(
+        '--marker',
+        help='marker type',
+        type=str,
+        choices=[x.value for x in Marker])
+
+    parser.add_argument('--save-image', help='save image', type=str)
+    parser.add_argument('--depth', help='depth', type=float)
+    parser.add_argument('--save-data', help='save data', type=str)
+    parser.add_argument('--save-config', help='save config', type=str)
+    args = parser.parse_args()
+
+    if args.version:
```

Comment:
`--version` flag should only return version number.
```shell
> samila --version
1.3
```

Add another flag like `--info` for this purpose.